sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileSystemClient.cs

```diff
@@ -889,19 +833,17 @@ private void SetNameFieldsIfNull()
         /// a failure occurs.
         /// </remarks>
         public virtual Response<FileSystemInfo> CreateIfNotExists(
-            Models.PublicAccessType publicAccessType = Models.PublicAccessType.None,
-            Metadata metadata = default,
-            DataLakeFileSystemEncryptionScopeOptions encryptionScopeOptions = default,
+            DataLakeFileSystemCreateOptions options,
             CancellationToken cancellationToken = default)
         {
             DiagnosticScope scope = ClientConfiguration.ClientDiagnostics.CreateScope($"{nameof(DataLakeFileSystemClient)}.{nameof(CreateIfNotExists)}");
             try
             {
                 scope.Start();
                 Response<BlobContainerInfo> containerResponse = _containerClient.CreateIfNotExists(
-                    publicAccessType: (Blobs.Models.PublicAccessType)publicAccessType,
-                    metadata: metadata,
-                    encryptionScopeOptions: encryptionScopeOptions.ToBlobContainerEncryptionScopeOptions(),
+                    publicAccessType: (Blobs.Models.PublicAccessType)options?.PublicAccessType,
+                    metadata: options?.Metadata,
+                    encryptionScopeOptions: options?.EncryptionScopeOptions.ToBlobContainerEncryptionScopeOptions(),
```

Comment:
can EncryptionScopeOptions be null ?

Reply:
Yes, but `ToBlobContainerEncryptionScopeOptions()` can handle the null case.

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/23130/files?file-filters%5B%5D=.cs#diff-fd396ffe12e6789625aa6b44879d8fe1a3e7a94ae915b2fa5dc09c23101d1e93R838